src/ExtraSpace.asm

```diff
@@ -203,7 +203,8 @@ SKIP_Y9 equ 0
 @ov36AlreadyLoaded:
 	.word 0 ; Set to 1 after loading our extra overlay. Has to be a word so we can directly load it with an ldr.
 @loadOverlay36:
-	sub sp,sp,10h
+	push lr
+	sub sp,sp,3ch
```

Comment:
`fn_getOverlayData` was actually corrupting the stack, so I had to increase the space. I guess it didn't crash at the old hook because the function never returned.

Reply:
Wait, I didn't realize you had added the `sub sp` here. Unless I'm visualizing the stack wrong, this should have no effect at all.

Reply:
You're right, everything seems to work fine without the stack allocation here. I don't really get it though, `fn_getOverlayData` writes the result to the stack address pointed to by `r0`, so I thought it should overwrite existing data on the stack?

Reply:
Yes, but the space required to write that result has already been allocated by `NitroMain`.

Is everything fixed now? No corruption or anything?

Reply:
I've tested it again with `MoveShortcuts` and everything seems good.